tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,339 @@
+
+	scanName = namegen.AppendRandomString("scan")
+
+	c.T().Log("Running a scan")
+
+	// Run the scan
+	err = runScan(client, scanName, c.cisProfileName, c.project.ClusterID)
+	require.NoError(c.T(), err)
+}
+
+func getClusterScanTemplate() (string, error) {
+	readYamlFile, err := os.ReadFile("./resources/cis-profiles/cluster-scan-template.yaml")
+	return string(readYamlFile), err
+}
+
+func findScanIdx(data []v1.SteveAPIObject, scanName string) int {
+	for idx, scanReport := range data {
+		if scanReport.Name == scanName {
+			return idx
+		}
+	}
+	return -1
+}
+
+func checkScanReport(steveClient *v1.Client, scanName string) error {
+	scanReportList, err := steveClient.SteveType(ClusterScanReportType).List(nil)
+	if err != nil {
+		return err
+	}
+
+	scanReportIdx := findScanIdx(scanReportList.Data, scanName)
+
+	for idx, scanReport := range scanReportList.Data {
+		if strings.Contains(scanReport.Name, scanName) {
+			scanReportIdx = idx
+			break
+		}
+	}
+
+	if scanReportIdx < 0 {
+		return errors.New("scan report not found")
+	}
+
+	cisReportSpec := &ClusterScanReportSpec{}
+	scanReportSpec := scanReportList.Data[scanReportIdx].Spec
+	err = v1.ConvertToK8sType(scanReportSpec, cisReportSpec)
+	if err != nil {
+		return err
+	}
+
+	reportData := &CisReport{}
+	err = json.Unmarshal([]byte(cisReportSpec.ReportJson), &reportData)
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Out of total number of %d scans, %d scans passed, %d scans were skipped and %d scans failed", reportData.Total, reportData.Pass, reportData.Skip, reportData.Fail)
+	for _, group := range reportData.Results {
+		for _, check := range group.Checks {
+			if check.State == Fail {
+				logrus.Infof("check failed: id: %s state: %s description: %s ", check.Id, check.State, check.Description)
+			}
+		}
+	}
+	if reportData.Fail != 0 {
+		return errors.New("cluster scan failed")
+	}
+	return nil
+}
+
+func runScan(client *rancher.Client, scanName, profileName, clusterID string) error {
+	clusterScanTemplate, err := getClusterScanTemplate()
+	if err != nil {
+		return err
+	}
+
+	yamlInput := &management.ImportClusterYamlInput{
+		DefaultNamespace: charts.CISBenchmarkNamespace,
+		YAML:             fmt.Sprintf(clusterScanTemplate, scanName, profileName),
+	}
+
+	// Get the cluster by ID using the client
+	cluster, err := client.Management.Cluster.ByID(clusterID)
+	if err != nil {
+		return err
+	}
+
+	// Import YAML into the cluster
+	_, err = client.Management.Cluster.ActionImportYaml(cluster, yamlInput)
+	if err != nil {
+		return err
+	}
+
+	// Get a client for interacting with resources in the cluster
+	steveClient, err := client.Steve.ProxyDownstream(clusterID)
+	if err != nil {
+		return err
+	}
+
+	// Poll until the scan is completed
+	err = wait.Poll(defaults.FiveHundredMillisecondTimeout, defaults.FiveMinuteTimeout, func() (done bool, err error) {
+		scanList, err := steveClient.SteveType(ClusterScanResourceType).List(nil)
```

Comment:
instead of listing all scans, is there a better way to do this? Like could you get the value from when you create the scan, or sort the list, maybe the scan has a query you can enter that sorts by the clusterName or even better, the scanName. 

this would reduce complexity later on, i.e. remove the need for findScanIDx